src/content/utils.ts

```diff
@@ -176,6 +176,18 @@ export async function extractSection(element: Element) {
 
   // ~~~ End of stupidly hacky code ~~~
 
+  // Extracting instructor details from the labels
+  const instructorLabels = element.parentElement?.querySelectorAll('[data-automation-id="promptOption"]');
```

Comment:
I think if we change to also look for not:(role = link) it will differentiate between the other ones and just return the instructors,  allowing us to make the loop less complicated.  Could you maybe try this, and see how it goes, thanks :) 

```  
// Extracting instructors. Query all the "promptOption" that are not links. If it has role as link, it is a Section Detail div, otherwise it is an Instructor div
  const instructorElements = element.parentElement?.querySelectorAll('[data-automation-id="promptOption"]:not([role="link"])');
  let instructors: string[] = [];
  
  instructorElements?.forEach((elm) => {
    instructors.push(elm?.textContent || "")
  })
```

Reply:
Good point! tried that and nothing broke with all the classes I tried (no instructor, 1 instructor, multiple instructor). Pushed change